### PR TITLE
Queries and update the animals table

### PR DESCRIPTION
**Pull request by ankitt26**
In this branch, I show the implementation of querying and updating the animals table by making use of transactions to update and delete records. 

- [ ]  Inside a transaction:

  - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
  - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Verify that changes were made.
  - Commit the transaction.
  - Verify that changes persist after commit.
 
image

- [ ] query 2
- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

image

- [ ]  Inside a transaction:

  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
  
image

- [ ]  Write queries to answer the following questions:

  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
  
image

### CODE REVIEW

**Review by topeogunleye — COMMENTED** [review]
@ankitt26,
 - I have reviewed your `README` file and overall it is well-structured with the necessary sections. However, I noticed that the `key-features` and `future-features` sections are currently empty. Kindly consider providing content for these sections to provide more comprehensive information to the readers. This will make your README valid

**Review by topeogunleye — COMMENTED** [review]
Hi @ankitt26, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## 🚫 Invalid Code Review Request 🚫

Thank you for your recent pull request. While we appreciate your effort, we regret to inform you that your changes have been invalidated. Here's the summary of the reasons:

1. **Template Adaptation**: The Microverse Template for the README was not adapted to your project. Instead, you copied and pasted the template as it is, which does not reflect the specific details and requirements of your project. It is crucial to tailor the README to accurately represent your project and provide relevant information to users and contributors.

2. **Lack of Project-specific Information**: A README serves as a crucial document to provide an overview, instructions, and details about your project. By not adapting the template, it fails to provide project-specific information that is necessary for others to understand and contribute to your work effectively.

3. **Documentation Clarity**: README files should be clear, concise, and easy to follow. The direct copy-paste approach can result in inconsistencies and confusion, as the template may contain irrelevant sections or sections that do not align with your project's structure or purpose.

We recommend taking the following steps to resolve the issue:

1. **Adapt the Template**: Carefully review the Microverse Template and adapt it to your project's specific requirements, structure, and goals. Ensure that all sections accurately represent your project, such as its purpose, features, installation instructions, usage examples, and contribution guidelines.

2. **Remove Irrelevant Placeholders**: Remove any placeholders that are meant for only the author of the project. For Example: 
```
> Write a message to encourage readers to support your project
> Give credit to everyone who inspired your codebase.
> Add at least 2 questions new developers would ask when they decide to use your project.
```
4. **Ensure Consistency**: Ensure that the README is consistent in terms of formatting, styling, and grammar. Pay attention to details and make sure the document is easy to read and understand.

Please make the necessary adjustments and resubmit your pull request. We appreciate your understanding and look forward to seeing your improved README that accurately represents your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by ankitt26** [comment]
Thanks for reviewing, I made the changes in the Readme File.

File: data.sql [hunk 1]
```diff
@@ -6,3 +6,12 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg )
     ('Pikachu', '2021-01-07', 1, false, 15.04 ),
      ('Devimon', '2017-05-12', 5, true, 11 );
 
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES
+    ('Charmander', '2020-02-08', 0, false, -11),
+    ('Plantmon', '2021-11-15', 2, true, -5.7),
+    ('Squirtle', '1993-04-02', 3, false, -12.13),
+    ('Angemon', '2005-06-12', 1, false, -45),
```
Comment (zieeco) [inline data.sql]: - Hey, please, kindly double-check the boolean value, it's meant to be `true` and not `false`. Could we update this please?

**Comment by topeogunleye** [comment]
> Thanks for reviewing, I made the changes in the Readme File.

You

> Thanks for reviewing, I made the changes in the Readme File.

@ankitt26 You're welcome, I'm glad you made the changes in the Readme File. It looks much better now. 😊

Please submit your project again for review.👏

Is there anything else you need help with? I'm always happy to assist you. 😊

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @ankitt26,





Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice schema file. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (ankitt26) [reply data.sql]: I made the requested changes

**Comment by ankitt26** [comment]
Made the changes requested by the last reviewer and update the attached screenshots.

**Review by frank1738 — APPROVED** [review]
Hi @ankitt26 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ well written queries 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks for reviewing 😊.

**Pull request merged**